### PR TITLE
[enhancement](merge-on-write) Add more log info when delete bitmap correctness check failed

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

when the delete bitmap correectness check failed, it will print logs like below indicating the rowsets that should have been considered and the missing rowsets' ids.
```
W0816 15:30:04.328012 2939820 tablet.cpp:3717] [txn_id:-1][tablet_id:10130][max_version: 12] check delete bitmap correctness failed!
W0816 15:30:04.328351 2939820 tablet.cpp:3765] {
    "requied_rowsets": [
        "[12-12] 2 DATA OVERLAPPING 020000000000003ac949b9452c99e0ea6388ebcb762536b1 131.37 KB",
        "[11-11] 2 DATA OVERLAPPING 0200000000000034c949b9452c99e0ea6388ebcb762536b1 131.36 KB",
        "[10-10] 2 DATA OVERLAPPING 0200000000000038c949b9452c99e0ea6388ebcb762536b1 136.82 KB",
        "[9-9] 2 DATA OVERLAPPING 0200000000000035c949b9452c99e0ea6388ebcb762536b1 136.82 KB",
        "[8-8] 2 DATA OVERLAPPING 0200000000000029c949b9452c99e0ea6388ebcb762536b1 131.37 KB",
        "[7-7] 2 DATA OVERLAPPING 0200000000000020c949b9452c99e0ea6388ebcb762536b1 131.37 KB",
        "[6-6] 2 DATA OVERLAPPING 0200000000000028c949b9452c99e0ea6388ebcb762536b1 136.82 KB",
        "[5-5] 2 DATA OVERLAPPING 0200000000000025c949b9452c99e0ea6388ebcb762536b1 136.82 KB",
        "[4-4] 1 DATA NONOVERLAPPING 0200000000000021c949b9452c99e0ea6388ebcb762536b1 136.82 KB",
        "[3-3] 1 DATA NONOVERLAPPING 020000000000001bc949b9452c99e0ea6388ebcb762536b1 131.37 KB",
        "[2-2] 1 DATA NONOVERLAPPING 0200000000000013c949b9452c99e0ea6388ebcb762536b1 181.04 KB"
    ],
    "missing_rowsets": [
        "0200000000000021c949b9452c99e0ea6388ebcb762536b1",
        "0200000000000020c949b9452c99e0ea6388ebcb762536b1",
        "0200000000000034c949b9452c99e0ea6388ebcb762536b1",
        "0200000000000013c949b9452c99e0ea6388ebcb762536b1",
        "020000000000001bc949b9452c99e0ea6388ebcb762536b1",
        "0200000000000025c949b9452c99e0ea6388ebcb762536b1",
        "0200000000000038c949b9452c99e0ea6388ebcb762536b1",
        "0200000000000028c949b9452c99e0ea6388ebcb762536b1",
        "020000000000003ac949b9452c99e0ea6388ebcb762536b1",
        "0200000000000029c949b9452c99e0ea6388ebcb762536b1",
        "0200000000000035c949b9452c99e0ea6388ebcb762536b1"
    ]
}
F0816 15:30:04.328388 2939820 tablet.cpp:3766] Check failed: false check delete bitmap correctness failed!

```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.26 seconds
 stream load tsv:          542 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.6 seconds inserted 10000000 Rows, about 337K ops/s
 storage size: 17162028423 Bytes

File: be/src/olap/tablet.cpp [hunk 1]
```diff
@@ -1479,6 +1479,129 @@ void Tablet::get_compaction_status(std::string* json_result) {
...
+    format_str = ToStringFromUnixMillis(_last_base_compaction_success_millis.load());
+    base_success_value.SetString(format_str.c_str(), format_str.length(), root.GetAllocator());
+    root.AddMember("last base success time", base_success_value, root.GetAllocator());
+    rapidjson::Value full_success_value;
+    format_str = ToStringFromUnixMillis(_last_full_compaction_success_millis.load());
+    full_success_value.SetString(format_str.c_str(), format_str.length(), root.GetAllocator());
+    root.AddMember("last full success time", full_success_value, root.GetAllocator());
+
+    // print all rowsets' version as an array
+    rapidjson::Document versions_arr;
```
Comment (zhannngchen) [inline be/src/olap/tablet.cpp]: Log with json is too costive

File: be/src/olap/tablet.cpp [hunk 2]
```diff
@@ -3698,16 +3821,33 @@ void Tablet::_remove_sentinel_mark_from_delete_bitmap(DeleteBitmapPtr delete_bit
...
+                     << "] check delete bitmap correctness failed!";
+        std::string rowset_status_string;
+        if (with_meta_lock) {
+            get_compaction_status_without_lock(&rowset_status_string);
+        } else {
+            get_compaction_status(&rowset_status_string);
+        }
+        LOG(WARNING) << rowset_status_string;
+        LOG(WARNING) << "can't find sentinel mark in the following rowsets:";
+        for (const auto& rowsetid : missing_ids) {
```
Comment (zhannngchen) [inline be/src/olap/tablet.cpp]: use `get_rowset_by_ids`, and print all rowset's version and rowset_id is enough.

File: be/src/olap/tablet.h [hunk 3]
```diff
@@ -301,8 +301,17 @@ class Tablet : public BaseTablet {
     // caller should hold the _meta_lock before calling this method
     void generate_tablet_meta_copy_unlocked(TabletMetaSharedPtr new_tablet_meta) const;
 
+    void extract_rowsets_unlocked(std::vector<RowsetSharedPtr>& rowsets);
```
Comment (zhannngchen) [inline be/src/olap/tablet.h:304]: These methods should be private?

**Comment by bobhan1** [comment]
run buildall

File: be/src/olap/tablet.cpp [hunk 4]
```diff
@@ -1355,6 +1355,16 @@ std::vector<RowsetSharedPtr> Tablet::pick_candidate_rowsets_to_build_inverted_in
     return candidate_rowsets;
 }
 
+std::string Tablet::get_rowset_info_str(RowsetSharedPtr rowset, bool delete_flag) {
```
Comment (zhannngchen) [inline be/src/olap/tablet.cpp:1358]: should be private

**Comment by zhannngchen** [comment]
add example logs to the description of this PR

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.69 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162345189 Bytes

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.12 seconds
 stream load tsv:          542 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162313169 Bytes

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**